### PR TITLE
Harden STT cancellation watchdog

### PR DESCRIPTION
**Pull request by moona3k**
## What Changed
- Adds a bounded watchdog for active STT job cancellation.
- Makes shutdown and clear-model-cache quiesce paths return after a bounded drain timeout.
- Marks the shared scheduler runtime unhealthy after an uncooperative runtime timeout and rejects new STT work instead of reusing a possibly wedged FluidAudio/WhisperKit runtime.
- Adds XCTest coverage for caller cancellation, cache clear, and shutdown when the mock runtime ignores cancellation.

## Why
STTScheduler previously depended on the underlying STT runtime returning after task cancellation. If FluidAudio or WhisperKit ignored cancellation or wedged, callers and in-app maintenance flows could hang indefinitely.

## Validation
- `swift build --skip-update --target MacParakeetCore` via Xcode Swift toolchain: passed.
- Rebuilt `MacParakeetPackageTests.xctest` with Xcode XCTest search/link paths: passed build; SwiftPM runner itself is blocked by this host's /usr/bin/xcrun Xcode-license prompt.
- Direct Xcode `xctest -XCTest STTSchedulerTests .build/arm64-apple-macosx/debug/MacParakeetPackageTests.xctest`: 22 tests passed.

## Notes
No existing PR existed for this branch before publishing, so there were no prior GitHub review threads or comments to reply to.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling and cleanup when speech engine initialization fails during recording startup
  * Added timeout handling to prevent the speech scheduler from becoming unresponsive during engine switching or job cancellation

* **Chores**
  * Expanded test coverage for error handling and cancellation edge cases

[Review Change Stack](https://app.coderabbit.ai/change-stack/moona3k/macparakeet/pull/242)

### CODE REVIEW

File: Sources/MacParakeetCore/STT/STTScheduler.swift [hunk 1]
```diff
@@ -380,23 +423,94 @@ public actor STTScheduler: STTManaging, SpeechEngineRoutedTranscribing, SpeechEn
...
+        }
+
+        guard !runningJobs.isEmpty else { return true }
+
+        let start = ContinuousClock.now
+        while start.duration(to: .now) < runningJobCancellationTimeout {
+            if runningJobs.allSatisfy({ slotState(for: $0.slot).currentJob?.id != $0.id }) {
+                return true
+            }
+            try? await Task.sleep(for: .milliseconds(20))
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/MacParakeetCore/STT/STTScheduler.swift]: Addressed in 9bec2158. The drain loop now catches Task.sleep cancellation explicitly, breaks out instead of spinning, and fails closed if any cancelled STT job is still active. That preserves the bounded watchdog behavior without a cancelled quiesce task busy-looping until the timeout.

**Comment by moona3k** [comment]
Closing in favor of #246, which ships **just the observability** for the failure mode this PR was defending against.

## Why we're swapping approaches

This PR's design is solid — bounded watchdogs, sticky unhealthy state, exclusive-operation flags, fail-closed remediation. The reason for the swap isn't quality; it's **timing**.

We have no production data showing the wedge condition (FluidAudio / WhisperKit ignoring cancellation) actually occurs. Zero user reports of stuck dictation, zero crash signatures, zero telemetry signal — but we also have no telemetry that *could* see it. So the question \"is this a real problem?\" is currently unanswerable.

Shipping ~500 LOC of remediation as insurance against an unobserved failure mode is expensive. Even if the design is correct, it introduces:

- New state-machine combinatorics on a hot-path actor
- A new user-visible failure mode (\"Speech runtime is unavailable until you restart\") that didn't exist before
- A sticky unhealthy state that can't recover without process restart

If the wedge condition is rare or non-existent in practice, all of that is overhead defending against a phantom.

## What #246 does instead

Adds **only the watchdog detection** — no state machine, no behavior change, no new error messages. When a runtime call exceeds the timeout, it fires a `stt_runtime_unhealthy` telemetry event and the caller keeps waiting (existing behavior preserved). About 50 lines of additive code instead of 500.

That gives us the data this PR's design needs to be justified. Then:

- **If the event fires regularly in production** → revive this PR's design (it's already fully reviewed and battle-tested by tests). We'd rebase and merge as the data-justified remediation.
- **If the event never fires** → delete the watchdog probes from #246 too. We confirmed the failure mode isn't real and saved ~500 LOC.

## Why this PR stays useful

Keeping it closed-but-not-deleted preserves:

1. The full reasoning + design (this thread's review threads)
2. The test scaffolding for cancellation-ignoring runtimes
3. The state-machine architecture, if we need to revive it

Closing without merging.
